### PR TITLE
fix(tools): correct dev server not serving lightdom.css files

### LINKED ISSUE #2794: [bug] dev server no longer serves -lightdom.css
A change in https://github.com/patternfly/patternfly-elements/commit/9d68f3d60ed4b76b03b37d23173c0a8f7780547b unfortunately broke lightdom files from being served out of `elements/pf-jazz-hands/pf-jazz-hands-lightdom.css`.  The files are no longer  redirected from `element/jazz-hands/pf-jazz-hands-lightdom.css` to the prefixed  `pf-jazz-hands` folder.

To reproduce create a `-lightdom.css` file and link it in the demo:

`/elements/pf-tabs/pf-tabs-lightdom.css`
```
pf-tabs {
  background: rebeccapurple;
}
```

`/elements/pf-tabs/demo/pf-tabs.html`
```
<link rel="stylesheet" href="../pf-tabs-lightdom.css">
```

outputs:
```
Request URL:
http://localhost:8001/components/tabs/pf-tabs-lightdom.css
Request Method:
GET
Status Code:
404 Not Found
```

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1. Fix dev-server not serving lightdom.css files

Closes #2794 

## Testing Instructions

1. Follow instructions in #2794 ensure `lightdom.css` and `lightdom-shim.css` serve.
2. Launch dev server `npm run serve`

## Notes to Reviewers

1.

### CODE REVIEW

File: tools/pfe-tools/dev-server/plugins/pfe-dev-server.ts [hunk 1]
```diff
@@ -97,24 +97,20 @@ function getRouter(options: PfeDevServerInternalConfig) {
...
   const { componentSubpath } = options.site;
   const router = new Router()
+
+      // PFE TOOLING
       .get('/tools/pfe-tools/environment.js(.js)?', async ctx => {
         ctx.body = await makeDemoEnv(options.rootDir);
         ctx.type = 'application/javascript';
       })
 
-  // Redirect `components/jazz-hands/*-lightdom.css` to `elements/pf-jazz-hands/*-lightdom.css`
```
Comment (zeroedin) [inline tools/pfe-tools/dev-server/plugins/pfe-dev-server.ts:105]: I found this block to be unnecessary and a much more simple check in the components/:element/* dir => elements/:element/* dir later

File: tools/pfe-tools/dev-server/config.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ function normalizeOptions(options?: PfeDevServerConfigOptions) {
   config.watchFiles ??= '{elements,core}/**/*.{css,html}';
   config.litcssOptions ??= {
     include: /\.css$/,
-    exclude: /(((fonts|demo)|(demo\/.*))\.css$)|(.*(-lightdom.css$))/,
+    exclude: /(((fonts|demo)|(demo\/.*))\.css$)|(.*(-lightdom.css$))|(.*(-lightdom-shim.css$))/,
```
Comment (zeroedin) [inline tools/pfe-tools/dev-server/config.ts:60]: most likely possible to shorten this with another splat in the `-lightdom.css` check? I did an attempt and didn't seem like it wanted to work but my regex foo isn't as strong as i'd like. YMMV

File: tools/pfe-tools/dev-server/plugins/pfe-dev-server.ts [hunk 3]
```diff
@@ -129,33 +125,34 @@ function getRouter(options: PfeDevServerInternalConfig) {
...
-      .get(`/core/pfe-core/:splatPath*.js`, async ctx => {
-        ctx.redirect(`/core/pfe-core/${ctx.params.splatPath}.ts`);
-      })
-
-  // Redirect `elements/pf-jazz-hands/*.js` to `elements/pf-jazz-hands/*.ts` for requests not previously handled
-      .get(`/${elementsDir}/:element/:splatPath*.js`, async ctx => {
-        const { element, splatPath } = ctx.params;
-        if (element.startsWith(tagPrefix) && !splatPath.includes('/')) {
-          ctx.redirect(`/${elementsDir}/${element}/${splatPath}.ts`);
-        }
```
Comment (zeroedin) [inline tools/pfe-tools/dev-server/plugins/pfe-dev-server.ts:158]: found this block to be unnecessary given the `liveReloadTsChangesMiddleware` that was added, the attempt was the same, but i think the middleware might be a better place for this particular redirect.

File: tools/pfe-tools/dev-server/plugins/pfe-dev-server.ts [hunk 4]
```diff
@@ -97,24 +97,20 @@ function getRouter(options: PfeDevServerInternalConfig) {
...
-  // Redirect `components/jazz-hands/*-lightdom.css` to `elements/pf-jazz-hands/*-lightdom.css`
-  // NOTE: don't put subresources in /demo called `*-lightdom.css` , or this will break
-      .get(`/${componentSubpath}/:element/(demo/)?:fileName.css`, async (ctx, next) => {
-        const { element, fileName } = ctx.params;
-        if (!element.startsWith(tagPrefix) && fileName.includes('lightdom')) {
-          const prefixedElement = deslugify(element);
-          ctx.redirect(`/${elementsDir}/${prefixedElement}/${fileName}.css`);
-        } else {
-          return next();
-        }
```
Comment (zeroedin) [inline tools/pfe-tools/dev-server/plugins/pfe-dev-server.ts:114]: I found this block to be unnecessary and a much more simple check in the `components/:element/*`  => `elements/:element/*` block later down in the code

**Comment by bennypowers** [comment]
iirc there's a fix for this on staging/4.0.

Will check on Sunday

File: tools/pfe-tools/dev-server/config.ts (hunk 2 above)
Comment (bennypowers) [reply tools/pfe-tools/dev-server/config.ts:60]: shameless plug: https://github.com/bennypowers/nvim-regexplainer

```markdown
Either capture group 1:
  capture group 2:
    Either capture group 3:
      Either `fonts` or `demo` or capture group 4:
      `demo/`
      **ANY** (_>= 0x_)
  `.css`
  **END**, capture group 5:
  **ANY** (_>= 0x_)
  capture group 6:
    `-lightdom`
    **ANY**
    `css`
    **END**, or capture group 7:
  **ANY** (_>= 0x_)
  capture group 8:
    `-lightdom-shim`
    **ANY**
    `css`
    **END**
```
I think we can simplify this to:

```js
/**
 * Either `fonts/`, `demo/`, or `-lightdom`
 * non-capturing group (_optional_):
 *   `-shim`
 * `.css`
 * **END**
 */
/fonts\/|demo\/|-lightdom(?:-shim)?\.css$/
```

**Comment by bennypowers** [comment]
Thanks for filing this. I decided to put things in order a bit, so 
- I refactored the dev server plugin to import it's middlewares from purpose-oriented modules
- I split the router middlewares out into small, directed functions, one per case.
- Improved the inline documentation for the router functions, hopefully clarifying what each middleware does

I then tested with these steps:

1. add `/elements/accordion/pf-accordion-lightdom.css` which sets background to blue
2. add `/elements/accordion/pf-accordion-lightdom-shim.css` which sets border to pink
2. add `/elements/accordion/demo/components@2x.png` copied from `/brand`
3. link to both stylesheets and the image from `/elements/pf-accordion/demo/pf-accordion.html`
3. link to both stylesheets and the image from `/elements/pf-accordion/demo/bordered.html`
4. build and run the dev server
5. ensure that in both the accordion and bordered demos, the css applies and the image loads
6. kill the dev server
7. copy (not npm link, but copy) the contents of `tools/pfe-tools/dev-server` to rhds repo's node_modules
8. replace the fully qualified urls in the avatar demos with local demos, and populate those images
9. run the dev server in the rhds directory
10. ensure that the avatar, avatar color context, table, and table color context demos render as expected

File: docs/_data/importMap.cjs
Comment (bennypowers) [inline docs/_data/importMap.cjs:1]: resolve to merge after testing

**Review by bennypowers — APPROVED** [review]

File: tools/pfe-tools/dev-server/plugins/dev-server-router.ts
Comment (zeroedin) [inline tools/pfe-tools/dev-server/plugins/dev-server-router.ts:1]: Love the organization here and the comments, much easier to grok now.  Thank you for putting time into this @bennypowers it is much appreciate

**Pull request merged**